components/mapbox_input.tsx

```diff
@@ -0,0 +1,99 @@
+import Head from 'next/head';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import React, { useRef, useState, useEffect } from 'react';
+import mapboxgl, { AttributionControl } from 'mapbox-gl';
+import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder';
+
+interface MapBoxInputProps {
+  setIsSubmissionEnabled: (isSubmissionEnabled: boolean) => void;
+  zoomLevelThreshold: number;
+}
+
+export default function MapBoxInput({
+  setIsSubmissionEnabled,
+  zoomLevelThreshold,
+}: MapBoxInputProps) {
+  mapboxgl.accessToken =
+    'pk.eyJ1IjoiYWxlc3RlcjMiLCJhIjoiY2xlM3JwdDkwMDR6cjNvdGRpanZqZHd0ciJ9.ibQNGDwEE_Wc59LB2dhs9Q';
```

Comment:
We'll probably want to store this in an environment variable, despite it being public and fine to expose. You can put this in your `.env` and access it with `process.env.<ENV_VAR_NAME>`